plugin/pkg/managedseed/validator/admission.go

```diff
@@ -239,13 +240,83 @@ func (v *ManagedSeed) Admit(ctx context.Context, a admission.Attributes, o admis
+	case admission.Update:
+		oldManagedSeed, ok := a.GetOldObject().(*seedmanagement.ManagedSeed)
+		if !ok {
+			return apierrors.NewInternalError(errors.New("could not covert old resource into ManagedSeed object"))
+		}
+		errs, err := v.validateManagedSeedUpdate(oldManagedSeed, managedSeed, shoot)
+		if err != nil {
+			return err
+		}
+		allErrs = append(allErrs, errs...)
+	}
+
 	if len(allErrs) > 0 {
 		return apierrors.NewInvalid(gk, managedSeed.Name, allErrs)
 	}
 
 	return nil
 }
 
+func (v *ManagedSeed) validateManagedSeedCreate(managedSeed *seedmanagement.ManagedSeed, shoot *gardencore.Shoot) (field.ErrorList, error) {
+	allErrs := field.ErrorList{}
+
+	seedSpec, err := seedmanagementhelper.ExtractSeedSpec(managedSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	shootZones := helper.GetAllZonesFromShoot(shoot)
+
+	if len(seedSpec.Provider.Zones) > 0 && !sets.New[string](seedSpec.Provider.Zones...).Equal(shootZones) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "gardenlet", "config", "seedConfig", "spec", "provider", "zones"), seedSpec.Provider.Zones, fmt.Sprintf("seed provider zones must be equal to shoot zones (%v)", sets.List(shootZones))))
+	}
+
+	return allErrs, nil
+}
+
+func (v *ManagedSeed) validateManagedSeedUpdate(oldManagedSeed, newManagedSeed *seedmanagement.ManagedSeed, shoot *gardencore.Shoot) (field.ErrorList, error) {
+	allErrs := field.ErrorList{}
+	zonesFieldPath := field.NewPath("spec", "gardenlet", "config", "seedConfig", "spec", "provider", "zones")
+
+	oldSeedSpec, err := seedmanagementhelper.ExtractSeedSpec(oldManagedSeed)
+	if err != nil {
+		return nil, err
+	}
+	newSeedSpec, err := seedmanagementhelper.ExtractSeedSpec(newManagedSeed)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := admissionutils.ValidateZoneRemovalFromSeeds(oldSeedSpec, newSeedSpec, newManagedSeed.Name, v.shootLister, "ManagedSeed"); err != nil {
+		allErrs = append(allErrs, field.Forbidden(zonesFieldPath, "zones must not be removed while shoots are still scheduled onto seed"))
+	}
+
+	shootZones := helper.GetAllZonesFromShoot(shoot)
+	newZones := sets.New[string](newSeedSpec.Provider.Zones...).Difference(sets.New[string](oldSeedSpec.Provider.Zones...))
+
+	// Newly added zones should match the ones found in the shoot cluster.
+	// Zone names were allowed to deviate from the zones configured for shoot clusters, see https://github.com/gardener/gardener/commit/8d28452e7f718d0041fbe82eb83543e3a87ea8ad.
+	// Thus, we can only check added zones here.
+	if !shootZones.HasAll(newZones.UnsortedList()...) {
+		allErrs = append(allErrs, field.Invalid(zonesFieldPath, newZones.UnsortedList(), "added zones must match the zones configured in the ManagedSeed's shoot cluster"))
+	}
+
+	if len(newSeedSpec.Provider.Zones) > 0 && shootZones.Len() != len(newSeedSpec.Provider.Zones) {
+		allErrs = append(allErrs, field.Invalid(zonesFieldPath, newSeedSpec.Provider.Zones, "seed provider zones size must be equal to shoot zones"))
```

Comment:
Isn't a valid use case to allow ManagedSeed's Shoot to add new zones?
IIUC, this validation will fail update of ManagedSeed if Seed .spec.provider.zones != shoot zones.